### PR TITLE
.github/workflows: add welcome bot

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

To try to address #18563 I would like to add a welcome bot.
This should help give a bit more urgency to the first time contributors and show that is ok to ping people.

### Testing procedure

Ya... turns out it is not too easy.

### Issues/PRs references

Helps with #18563

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Testing "locally" on my repo doesn't seem to work out well. I currently am hitting https://github.com/actions/first-interaction/issues/10 issue.

[Here](https://github.com/MrKevinWeiss/RIOT/actions/workflows/welcome.yml) are my test logs.

The other thing that may be an issue is requiring maintainer approval to run this workflow.  This kind of makes a welcome message to be patient a bit worthless.  We can change the settings to run ALL workflows of first time contributors (unless they have a newly created account) without approval.

It would be better to configure it to only run this one though.

maybe ping @miri64 ?

**Comment by MrKevinWeiss** [comment]
General review.  I feel you are the most knowledgeable GitHub actions person I know.  If you are aware of a way we can test this then great, otherwise, does this make sense to merge in, as it is similar to the labeler workflow?

**Comment by miri64** [comment]
IIRC it is easiest to test Github Actions in your own fork. You can provide links to the actions / PRs you tested upon.

File: .github/workflows/welcome.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+    - uses: actions/first-interaction@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        issue-message: Beep boop beep... I am a real human... for real.\n\n
+          Welcome to the RIOT community, we will try to address this as soon as we can.
+          As I mentioned before, we are human and sometimes we get distracted with life.
```
Comment (miri64) [inline .github/workflows/welcome.yml:24]:
```suggestion
          As I mentioned before, we are human, many of us do this in our free time, and sometimes we get distracted with life.
```
?

File: .github/workflows/welcome.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          I can try to ping someone like this.\n\n
+          ping @MrKevinWeiss\n\n
+          Feel free to do the same for other [maintainers](https://github.com/orgs/RIOT-OS/teams/maintainers/members) if things are taking a while.
+        pr-message: Beep boop beep... I am a real human... for real.\n\n
+          Welcome to the RIOT community, we will try to review this as soon as we can.
+          As I mentioned before, we are human and sometimes we get distracted with life.
+          I can try to ping someone like this.\n\n
+          ping @MrKevinWeiss\n\n
+          Feel free to do the same for other [maintainers](https://github.com/orgs/RIOT-OS/teams/maintainers/members) if things are taking a while.
```
Comment (miri64) [inline .github/workflows/welcome.yml:33]: If this is the same message as for `issue-message`, could you try setting it as an environment variable and use it here and for `issue-message` instead?

```yaml
    - uses: actions/first-interaction@v1
      env:
        GREETER_MESSAGE: >
          Beep boop beep... I am a real human... for real.\n\n
          Welcome to the RIOT community, we will try to address this as soon as we can.
          As I mentioned before, we are human and sometimes we get distracted with life.
          I can try to ping someone like this.\n\n
          ping @MrKevinWeiss\n\n
          Feel free to do the same for other [maintainers](https://github.com/orgs/RIOT-OS/teams/maintainers/members) if things are taking a while.
      with:
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        issue-message: ${{ env.GREETER_MESSAGE }}
        pr-message: ${{ env.GREETER_MESSAGE }}
```

File: .github/workflows/welcome.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      pull-requests: write
+    steps:
+    - uses: actions/first-interaction@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        issue-message: Beep boop beep... I am a real human... for real.\n\n
+          Welcome to the RIOT community, we will try to address this as soon as we can.
+          As I mentioned before, we are human and sometimes we get distracted with life.
+          I can try to ping someone like this.\n\n
+          ping @MrKevinWeiss\n\n
```
Comment (miri64) [inline .github/workflows/welcome.yml:26]: You really want to use yourself as example? How about, e.g., @riot-ci instead?

**Comment by miri64** [comment]
> IIRC it is easiest to test Github Actions in your own fork. You can provide links to the actions / PRs you tested upon.

Mh, just noticed: the action ran, so I am not sure what you were asking ^^

**Comment by MrKevinWeiss** [comment]
Thanks for taking a look, I will adapt!

> Mh, just noticed: the action ran, so I am not sure what you were asking ^^

The problem is that it never worked (as well as is very hard to test first time contributors as I need a new contributor per test).

I thought maybe it is an issue of my personal repo settings or something.   I will keep looking at it when I have time.

**Comment by MrKevinWeiss** [comment]
Hmm OK, shall we or should I close, I am doing a general cleanup.

**Comment by miri64** [comment]
Doesn't seem to work, so I am for closing...

**Comment by MrKevinWeiss** [comment]
Good good. Wasn't so easy to test.

**Pull request closed without merging**